cdap-formats/src/main/java/io/cdap/cdap/format/io/StructuredRecordDatumReader.java

```diff
@@ -50,6 +52,17 @@ public StructuredRecord read(Decoder decoder, Schema sourceSchema) throws IOExce
    * @throws IOException If failed to decode
    */
   protected final Object decode(Decoder decoder, Schema schema) throws IOException {
+    //handle any logical types that need special handling
+    if (schema.getLogicalType() != null) {
+      switch (schema.getLogicalType()) {
+        case DATETIME:
+          try {
+            return LocalDateTime.parse(decoder.readString());
+          } catch (DateTimeParseException dateTimeParseException) {
+            throw new IOException("Unsupported format. DateTime type should be in ISO-8601 format.");
```

Comment:
nit: considering the specific format follows this convention: https://docs.oracle.com/javase/8/docs/api/java/time/format/DateTimeFormatter.html#ISO_LOCAL_DATE_TIME
should we display the expected format in the exception?

I think this format matches `yyyy-MM-ddTHH-mm-ss`

Reply:
Removed this code because of https://github.com/cdapio/cdap/pull/13089#discussion_r565508603